mithril-stm/src/schnorr_signature/signing_key.rs

```diff
@@ -0,0 +1,223 @@
+use anyhow::{Context, anyhow};
+use dusk_jubjub::{
+    ExtendedPoint as JubjubExtended, Fq as JubjubBase, Fr as JubjubScalar,
+    SubgroupPoint as JubjubSubgroup,
+};
+use dusk_poseidon::{Domain, Hash};
+use group::Group;
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{
+    StmResult,
+    schnorr_signature::{
+        DST_SIGNATURE, SchnorrSignature, SchnorrSignatureError, SchnorrVerificationKey,
+        generate_non_zero_scalar, get_coordinates_several_points,
+    },
+};
+
+/// Schnorr Signing key, it is essentially a random scalar of the Jubjub scalar field
+#[derive(Debug, Clone)]
+pub struct SchnorrSigningKey(pub(crate) JubjubScalar);
+
+impl SchnorrSigningKey {
+    /// Generate a random scalar value to use as signing key
+    pub fn generate<R: RngCore + CryptoRng>(rng: &mut R) -> StmResult<Self> {
```

Comment:
This is a better naming as this can fail:
```suggestion
    pub fn try_generate<R: RngCore + CryptoRng>(rng: &mut R) -> StmResult<Self> {
```